pygmt/datasets/earth_dist.py

```diff
@@ -0,0 +1,104 @@
+"""
+Function to download the GSHHG Earth distance to shoreline dataset from the GMT data
+server, and load as :class:`xarray.DataArray`.
+
+The grids are available in various resolutions.
+"""
+
+from collections.abc import Sequence
+from typing import Literal
+
+import xarray as xr
+from pygmt.datasets.load_remote_dataset import _load_remote_dataset
+
+__doctest_skip__ = ["load_earth_dist"]
+
+
+def load_earth_dist(
+    resolution: Literal[
+        "01d", "30m", "20m", "15m", "10m", "06m", "05m", "04m", "03m", "02m", "01m"
+    ] = "01d",
+    region: Sequence[float] | str | None = None,
+    registration: Literal["gridline", "pixel"] = "gridline",
+) -> xr.DataArray:
+    r"""
+    Load the GSHHG Earth distance to shoreline dataset in various resolutions.
+
+    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_earth_dist.jpg
+       :width: 80 %
+       :align: center
+
+       GSHHG Earth distance to shoreline dataset.
+
+    The grids are downloaded to a user data directory (usually
+    ``~/.gmt/server/earth/earth_dist/``) the first time you invoke this function.
+    Afterwards, it will load the grid from the data directory. So you'll need an
+    internet connection the first time around.
+
+    These grids can also be accessed by passing in the file name
+    **@earth_dist**\_\ *res*\[_\ *reg*] to any grid processing function or plotting
+    method. *res* is the grid resolution (see below), and *reg* is the grid registration
+    type (**p** for pixel registration or **g** for gridline registration).
+
+    The default color palette table (CPT) for this dataset is *@earth_dist.cpt*. It's
+    implicitly used when passing in the file name of the dataset to any grid plotting
+    method if no CPT is explicitly specified. When the dataset is loaded and plotted
+    as an :class:`xarray.DataArray` object, the default CPT is ignored, and GMT's
+    default CPT (*turbo*) is used. To use the dataset-specific CPT, you need to
+    explicitly set ``cmap="@earth_dist.cpt"``.
+
+    Refer to :gmt-datasets:`earth-dist.html` for more details about available datasets,
+    including version information and references.
+
+    Parameters
+    ----------
+    resolution
+        The grid resolution. The suffix ``d`` and ``m`` stand for arc-degrees and
+        arc-minutes.
+    region
+        The subregion of the grid to load, in the form of a sequence [*xmin*, *xmax*,
+        *ymin*, *ymax*] or an ISO country code. Required for grids with resolutions
+        higher than 5 arc-minutes (i.e., ``"05m"``).
+    registration
+        Grid registration type. Either ``"pixel"`` for pixel registration or
+        ``"gridline"`` for gridline registration.
+
+    Returns
+    -------
+    grid
+        The GSHHG Earth distance to shoreline grid. Coordinates are latitude and
+        longitude in degrees. Distances are in kilometers.
```

Comment:
```suggestion
        The GSHHG Earth distance to shoreline grid. Coordinates are latitude and
        longitude in degrees. Distances are in kilometers. Positive (negative) 
        distance values mean land to coastline (ocean to coastline).
```

Reply:
See commit https://github.com/GenericMappingTools/pygmt/pull/3706/commits/0682a1042b0bb123d2d092dc5da5a7536576aa98 for a slightly modified formulation.